pypeit/spectrographs/keck_mosfire.py

```diff
@@ -432,3 +516,233 @@ def tweak_standard(self, wave_in, counts_in, counts_ivar_in, gpm_in, meta_table,
+        objects = np.array([np.array(ssl['Slit_Number'], dtype=int),
+                           np.zeros(ssl['Slit_Number'].size, dtype=int),   # no object ID
+                           obj_ra,
+                           obj_dec,
+                           objname,
+                           np.array(targs_mtch['Magnitude'], dtype=float),
+                           ['None']*ssl['Slit_Number'].size,       # no magnitude band
+                           topdist,
+                           botdist]).T
+
+        # PA corresponding to positive x on detector (spatial)
+        posx_pa = hdu[0].header['SKYPA2']
+        if posx_pa < 0.:
+            posx_pa += 360.
+
+        slit_ra = ssl['Slit_RA_Hours']+' '+ssl['Slit_RA_Minutes']+' '+ssl['Slit_RA_Seconds']
+        slit_dec = ssl['Slit_Dec_Degrees']+' '+ssl['Slit_Dec_Minutes']+' '+ssl['Slit_Dec_Seconds']
+        slit_ra, slit_dec = meta.convert_radec(slit_ra, slit_dec)
+
+        # Instantiate the slit mask object and return it
+        self.slitmask = SlitMask(np.array([np.zeros(ssl['Slit_Number'].size),   # mosfire maskdef has not slit corners
+                                           np.zeros(ssl['Slit_Number'].size),
+                                           np.zeros(ssl['Slit_Number'].size),
+                                           np.zeros(ssl['Slit_Number'].size),
+                                           np.zeros(ssl['Slit_Number'].size),
+                                           np.zeros(ssl['Slit_Number'].size),
+                                           np.zeros(ssl['Slit_Number'].size),
+                                           np.zeros(ssl['Slit_Number'].size)]).T.reshape(-1,4,2),
+                                 slitid=np.array(ssl['Slit_Number'], dtype=int),
+                                 align=ssl['Target_Name'] == 'posB',
+                                 science=ssl['Target_Name'] != 'posB',
+                                 onsky=np.array([slit_ra,
+                                                 slit_dec,
+                                                 np.array(ssl['Slit_length'], dtype=float),
+                                                 np.array(ssl['Slit_width'], dtype=float),
+                                                 np.array([round(hdu[0].header['SKYPA3'],2)]*ssl['Slit_Number'].size)]).T,
+                                 objects=objects,
+                                 posx_pa=posx_pa)
+        return self.slitmask
+
+    def get_maskdef_slitedges(self, ccdnum=None, filename=None, debug=None):
+        """
+         Provides the slit edges positions predicted by the slitmask design using
+         the mask coordinates already converted from mm to pixels by the method
+        `mask_to_pixel_coordinates`.
+
+        If not already instantiated, the :attr:`slitmask`, :attr:`amap`,
+        and :attr:`bmap` attributes are instantiated.  If so, a file must be provided.
+
+        Args:
+        ccdnum (:obj:`int`):
```

Comment:
the variables may need to be indented for sphinx to work right.  not sure.

Reply:
Yes! Thanks. Done